web/src/app/forms/Form.js

```diff
@@ -9,42 +9,40 @@ import { FormContext } from './context'
  *
  * onSubmit (if provided) will be called with a second `isValid` argument.
  */
-export class Form extends React.PureComponent {
-  static propTypes = {
-    onSubmit: p.func,
-    disabled: p.bool,
-  }
-
-  _checks = []
+export function Form(props) {
+  let _checks = []
```

Comment:
`_checks` as defined will reset to `[]` every time `Form` rerenders. `_checks` was originally bound to `this` within the class, which will retain its value across renders (acting like a synchronous setState)

Reply:
https://reactjs.org/docs/hooks-reference.html#useref
```
const checks = useRef([]) // gives us {current: []} that stays the same each render

// checks.current = checks.current.filter(...)
```

even though `useState` works synchronously, it would cause a render every time this changes, which we would not want